sirepo/package_data/static/js/elegant.js

```diff
@@ -1443,3 +1443,81 @@ SIREPO.app.directive('srBunchEditor', function(appState, panelState) {
         },
     };
 });
+
+
+SIREPO.app.directive('viewLogIframe', function(appState, requestSender) {
+    return {
+        restrict: 'A',
+        scope: {
+
+        },
+        // TODO: At error case for when log cannot be found, edit case below
+        template: `
+            <a href data-ng-click="viewLog()">View Log</a>
+            <div class="modal fade" id="sr-iframe-text-view" tabindex="-1" role="dialog">
+              <div class="modal-dialog modal-lg">
+                <div class="modal-content">
+                  <div class="modal-header bg-warning">
+                    <span class="lead modal-title text-info">Log</span>
+                    <div class="sr-panel-options pull-right">
+                      <a data-ng-href="{{ downloadLog() }}" target="_blank">
+                        <span class="sr-panel-heading glyphicon glyphicon-cloud-download" style="margin-bottom: 0"></span>
+                      </a>
+                      &nbsp&nbsp
+                      <button type="button" class="close" data-dismiss="modal">
+                        <span>&times;</span>
+                      </button>
+                    </div>
+                  </div>
+                  <div class="modal-body" style="padding: 0">
+                    <iframe id="sr-text-iframe"
+                      style="border: 0; width: 100%; height: 80vh" src=""></iframe>
+                  </div>
+                </div>
+              </div>
+            </div>
+        `,
+        controller: function(appState, requestSender, $scope) {
+
+            function setIFrameHTML(html) {
+                $('#sr-text-iframe').contents().find('html').html(html);
+            }
+
+            function getDownloadURL(model, index) {
+                if (! appState.isLoaded() || ! model) {
+                    return '';
+                }
+                return requestSender.formatUrl('downloadDataFile', {
+                    '<simulation_id>': appState.models.simulation.simulationId,
+                    '<simulation_type>': SIREPO.APP_SCHEMA.simulationType,
+                    '<model>': model,
+                    '<frame>': index,
+                });
+            }
+
+            $scope.viewLog = function(model, filename) {
+                if (! appState.isLoaded()) {
+                    return '';
+                }
+                
+                requestSender.sendStatelessCompute(
+                    appState,
+                    (data) => {
+                        appState.models.modelLog = data.html;
+                        setIFrameHTML(appState.models.modelLog);
+                    },
+                    {
+                        method: 'log_to_html',
+                        model: appState.models.simulationStatus.animation.computeModel,
+                        filename: 'elegant.log',
+                        simulationId: appState.models.simulation.simulationId
+                    });
```

Comment:
Don't pass filename to the server - that would allow this API to access any file on the server. Instead, the server side should use the ELEGANT_LOG_FILE constant for the filename.